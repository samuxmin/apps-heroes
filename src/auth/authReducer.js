import { types } from "../types/types";

// const state = {
//     name:'Samuel',
//     logged:false
// }

export const authReducer = (state = {}, action) => {
   
    switch (action.type) {
        case types.login:
            return{
                ...action.payload, 
                logged:true
            }
        case types.logout:
            return{
                logged:false
            }

        default:
            return state;
    }
}
