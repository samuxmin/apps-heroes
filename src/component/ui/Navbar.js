import React, { useContext } from "react";
import { Link, NavLink, useHistory } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types/types";

export const Navbar = () => {
  const { user, dispatch } = useContext(AuthContext);
  const history = useHistory();

  const handleLogout = () => {
    history.replace("/login");
    dispatch({ type: types.logout });
  };
  return (
    <nav className="navbar-container">
      <Link className="nav-item" to="/">
        ALL
      </Link>

      <NavLink activeclassname="active" className="nav-item" exact to="/marvel">
        MARVEL
      </NavLink>

      <NavLink activeclassname="active" className="nav-item" exact to="/dc">
        DC
      </NavLink>

      <NavLink activeclassname="active" className="nav-item" exact to="/search">
        Search
      </NavLink>
      <button
        activeclassname="active"
        className="nav-item right logout-btn"
        onClick={handleLogout}
      >
        Logout
      </button>
      <span className="nav-item right">{user.name}</span>
    </nav>
  );
};
