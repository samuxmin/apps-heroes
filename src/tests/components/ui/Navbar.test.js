import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { Navbar } from "../../../component/ui/Navbar";
import { AuthContext } from "../../../auth/AuthContext";
import { MemoryRouter, Router } from "react-router-dom";
import { types } from "../../../types/types";

describe("Pruebas en <Navbar />", () => {
  const historyMock = {
    replace: jest.fn(),
    push: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
  };
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      name: "sam",
      logged: true,
    },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter>
        <Router history={historyMock}>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );
  afterAll(() => {
    jest.clearAllMocks();
  });
  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("span").text().trim()).toBe(contextValue.user.name);
  });

  test("debe de llamar el logout y usar history", () => {
    wrapper.find("button").prop("onClick")();
    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.logout,
    });
    expect(historyMock.replace).toHaveBeenCalledWith("/login");
  });
});
