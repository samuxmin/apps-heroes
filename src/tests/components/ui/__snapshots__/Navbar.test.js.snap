// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> debe de mostrarse correctamente 1`] = `
<nav
  className="navbar-container"
>
  <a
    className="nav-item"
    onClick={[Function]}
  >
    ALL
  </a>
  <a
    activeclassname="active"
    aria-current={null}
    className="nav-item"
    onClick={[Function]}
  >
    MARVEL
  </a>
  <a
    activeclassname="active"
    aria-current={null}
    className="nav-item"
    onClick={[Function]}
  >
    DC
  </a>
  <a
    activeclassname="active"
    aria-current={null}
    className="nav-item"
    onClick={[Function]}
  >
    Search
  </a>
  <button
    activeclassname="active"
    className="nav-item right logout-btn"
    onClick={[Function]}
  >
    Logout
  </button>
  <span
    className="nav-item right"
  >
    sam
  </span>
</nav>
`;
